vapi/esx/settings/clusters/vms/transition.go

```diff
@@ -0,0 +1,226 @@
+//   - ServiceUnavailable: If the service is not available.
+//   - Unauthenticated: If the caller is not authenticated.
+//   - Unauthorized: If the user doesn't have the required privileges.
+func (m *Manager) Enable(ctx context.Context, cluster types.ManagedObjectReference, solution string, spec *EnableSpec) error {
+	p := clusterTransitionPath(cluster).String()
+	url := m.Resource(p).WithSubpath(solution).WithParam("action", "enable").WithParam("vmw-task", "true")
+	var taskId string
+
+	if err := m.Do(ctx, url.Request(http.MethodPost, spec), &taskId); err != nil {
+		return err
+	}
+
+	_, err := tasks.NewManager(m.Client).WaitForRunningOrError(ctx, taskId)
+	return err
+}
+
+// MultiSourceEnable enables multiple EAM managed solutions in vLCM as a single solution. The
+// solution specification is validated before the enablement is started.
+//
+// The enablement only transfers ownership of the solutions from EAM to LCCM
+// and sets the desired state in LCCM. The new desired state is not applied,
+// the solution system VMs are untouched.
+//
+// The following happens once the operation is started:
+//   - A removal of the corresponding agencies in EAM is triggered.
+//
+// The following happens once the operation is completed:
+//   - The corresponding agencies in EAM can no longer be controlled through the EAM API.
+//   - The management of the desired solution specification can be done only through vLCM. See Solutions
+//
+// Supported only for solutions with deployment type DeploymentType#CLUSTER_VM_SET.
+//
+// Parameters:
+//   - cluster: Identifier of the cluster.
+//   - solution: Identifier of the solution.
+//   - spec: Enablement specification.
+//
+// Returns an error if:
+//   - There is an internal error. The accompanying error message will give more details about the failure.
+//   - InvalidArgument: If the validation of the cluster desired specification fails, the solution associated with solution already exists in vLCM for the cluster, or the agencies associated with the eamAgencyIDs exist, but are not in the scope of the cluster. The value of the data field of Error contains more information. It is a structure that contains all the fields defined in ValidationResult.
+//   - NotAllowedInCurrentState: If System VMs are disabled on the cluster via internal ESX Agent Manager (EAM) API (eam.EsxtAgentManager#disable).
+//   - Unsupported: If the cluster associated with cluster is not managed by vLCM.
+//   - NotFound: If there is no cluster associated with the cluster or if there is no EAM agency associated with any of the IDs in eamAgencyIDs field of the spec in the system.
+//   - ServiceUnavailable: If the service is not available.
+//   - Unauthenticated: If the caller is not authenticated.
+//   - Unauthorized: If the user doesn't have the required privileges.
+func (m *Manager) MultiSourceEnable(ctx context.Context, cluster types.ManagedObjectReference, solution string, spec *MultiSourceEnableSpec) error {
```

Comment:
Make this a private function, ex. `func (m *Manager) multiSourceEnable` with two public version:

* `func (m *Manager) MultiSourceEnable`
* `func (m *Manager) MultiSourceEnableAsync`

They both call the private one, but the second one returns the task instead of blocking.